pootle/apps/pootle_format/formats/po.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from translate.storage import poheader
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.utils import timezone
+
+from pootle.core.mixins import CachedMethods
+from pootle.core.utils import dateformat
+from pootle.core.utils.timezone import datetime_min
+from pootle.core.utils.version import get_major_minor_version
+from pootle_statistics.models import Submission
+from pootle_store.syncer import StoreSyncer
+
+
+class PoStoreSyncer(StoreSyncer):
+
+    def get_latest_submission(self, mtime):
+        user_displayname = None
+        user_email = None
+        fields = (
+            "submitter__username",
+            "submitter__full_name",
+            "submitter__email")
+        submissions = self.translation_project.submission_set.exclude(
+            submitter__username="nobody")
+        try:
+            username, fullname, user_email = (
+                submissions.filter(creation_time=mtime)
+                           .values_list(*fields).latest())
+        except Submission.DoesNotExist:
+            try:
+                _mtime, username, fullname, user_email = (
+                    submissions.values_list("creation_time", *fields)
+                               .latest())
+                mtime = min(_mtime, mtime)
+            except ObjectDoesNotExist:
+                pass
+        if user_email:
+            user_displayname = (
+                fullname.strip()
+                if fullname.strip()
+                else username)
+        return mtime, user_displayname, user_email
+
+    def get_po_revision_date(self, mtime):
```

Comment:
tbh - i think they should probs get factored out at some stage - but yep i would rather keep it clear that this is not part of the default format sync